## Observations and Insights

## Dependencies and starter code

In [1]:
%matplotlib notebook

In [26]:
# Dependencies and Setup
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as st
import numpy as np


# Study data files
mouse_metadata = "data/Mouse_metadata.csv"
study_results = "data/Study_results.csv"

In [3]:
# Read the mouse data and the study results
mouse_metadata = pd.read_csv(mouse_metadata)
mouse_metadata

Mouse ID Drug Regimen     Sex  Age_months  Weight (g)
0       k403     Ramicane    Male          21          16
1       s185    Capomulin  Female           3          17
2       x401    Capomulin  Female          16          15
3       m601    Capomulin    Male          22          17
4       g791     Ramicane    Male          11          16
..       ...          ...     ...         ...         ...
244     z314     Stelasyn  Female          21          28
245     z435     Propriva  Female          12          26
246     z581    Infubinol  Female          24          25
247     z795     Naftisol  Female          13          29
248     z969     Naftisol    Male           9          30

[249 rows x 5 columns]

In [4]:
len(mouse_metadata)

249

In [5]:
# Read the mouse data and the study results
study_results = pd.read_csv(study_results)
study_results

Mouse ID  Timepoint  Tumor Volume (mm3)  Metastatic Sites
0        b128          0           45.000000                 0
1        f932          0           45.000000                 0
2        g107          0           45.000000                 0
3        a457          0           45.000000                 0
4        c819          0           45.000000                 0
...       ...        ...                 ...               ...
1888     r944         45           41.581521                 2
1889     u364         45           31.023923                 3
1890     p438         45           61.433892                 1
1891     x773         45           58.634971                 4
1892     b879         45           72.555239                 2

[1893 rows x 4 columns]

In [6]:
len(study_results)

1893

There are total of 249 mouse subjected for testing. Each mouse has a tumore that was measured per five days. Timepoint represents the day number and Tumor Volume represent the tumor size.

In [7]:
# Combine the data into a single dataset

merge_data = pd.merge(mouse_metadata, study_results, on="Mouse ID", how="outer")
merge_data

Mouse ID Drug Regimen   Sex  Age_months  Weight (g)  Timepoint  \
0        k403     Ramicane  Male          21          16          0   
1        k403     Ramicane  Male          21          16          5   
2        k403     Ramicane  Male          21          16         10   
3        k403     Ramicane  Male          21          16         15   
4        k403     Ramicane  Male          21          16         20   
...       ...          ...   ...         ...         ...        ...   
1888     z969     Naftisol  Male           9          30         25   
1889     z969     Naftisol  Male           9          30         30   
1890     z969     Naftisol  Male           9          30         35   
1891     z969     Naftisol  Male           9          30         40   
1892     z969     Naftisol  Male           9          30         45   

      Tumor Volume (mm3)  Metastatic Sites  
0              45.000000                 0  
1              38.825898                 0  
2              35.014271                 1  
3              34.223992                 1  
4              32.997729                 1  
...                  ...               ...  
1888           63.145652                 2  
1889           65.841013                 3  
1890           69.176246                 4  
1891           70.314904                 4  
1892           73.867845                 4  

[1893 rows x 8 columns]

In [8]:
merge_data = merge_data[["Mouse ID", "Timepoint", "Drug Regimen", "Sex", "Age_months", "Weight (g)", "Tumor Volume (mm3)", "Metastatic Sites"]]
merge_data

Mouse ID  Timepoint Drug Regimen   Sex  Age_months  Weight (g)  \
0        k403          0     Ramicane  Male          21          16   
1        k403          5     Ramicane  Male          21          16   
2        k403         10     Ramicane  Male          21          16   
3        k403         15     Ramicane  Male          21          16   
4        k403         20     Ramicane  Male          21          16   
...       ...        ...          ...   ...         ...         ...   
1888     z969         25     Naftisol  Male           9          30   
1889     z969         30     Naftisol  Male           9          30   
1890     z969         35     Naftisol  Male           9          30   
1891     z969         40     Naftisol  Male           9          30   
1892     z969         45     Naftisol  Male           9          30   

      Tumor Volume (mm3)  Metastatic Sites  
0              45.000000                 0  
1              38.825898                 0  
2              35.014271                 1  
3              34.223992                 1  
4              32.997729                 1  
...                  ...               ...  
1888           63.145652                 2  
1889           65.841013                 3  
1890           69.176246                 4  
1891           70.314904                 4  
1892           73.867845                 4  

[1893 rows x 8 columns]

In [9]:
#Create main df

main_df = merge_data
main_df.sort_values("Mouse ID", inplace=True)
main_df.set_index(["Mouse ID","Timepoint"], inplace=True)
main_df.head(20)


Drug Regimen     Sex  Age_months  Weight (g)  \
Mouse ID Timepoint                                                
a203     30           Infubinol  Female          20          23   
         35           Infubinol  Female          20          23   
         25           Infubinol  Female          20          23   
         20           Infubinol  Female          20          23   
         15           Infubinol  Female          20          23   
         10           Infubinol  Female          20          23   
         40           Infubinol  Female          20          23   
         45           Infubinol  Female          20          23   
         5            Infubinol  Female          20          23   
         0            Infubinol  Female          20          23   
a251     20           Infubinol  Female          21          25   
         30           Infubinol  Female          21          25   
         35           Infubinol  Female          21          25   
         5            Infubinol  Female          21          25   
         40           Infubinol  Female          21          25   
         15           Infubinol  Female          21          25   
         10           Infubinol  Female          21          25   
         25           Infubinol  Female          21          25   
         45           Infubinol  Female          21          25   
         0            Infubinol  Female          21          25   

                    Tumor Volume (mm3)  Metastatic Sites  
Mouse ID Timepoint                                        
a203     30                  59.523197                 1  
         35                  61.931650                 2  
         25                  56.793208                 1  
         20                  55.173336                 1  
         15                  52.777870                 1  
         10                  51.852437                 1  
         40                  63.593490                 2  
         45                  67.973419                 2  
         5                   48.508468                 0  
         0                   45.000000                 0  
a251     20                  54.462594                 1  
         30                  59.861241                 1  
         35                  60.903885                 1  
         5                   46.982127                 0  
         40                  63.095076                 1  
         15                  50.534456                 1  
         10                  48.591179                 1  
         25                  57.414646                 1  
         45                  65.525743                 1  
         0                   45.000000                 0

## Summary statistics

In [10]:
# Generate a summary statistics table of mean, median, variance, standard deviation, and SEM of the 
# tumor volume for each regimen

In [11]:
stats_df = main_df[["Drug Regimen", "Tumor Volume (mm3)"]]
stats_df

Drug Regimen  Tumor Volume (mm3)
Mouse ID Timepoint                                 
a203     30           Infubinol           59.523197
         35           Infubinol           61.931650
         25           Infubinol           56.793208
         20           Infubinol           55.173336
         15           Infubinol           52.777870
...                         ...                 ...
z969     20            Naftisol           57.898778
         25            Naftisol           63.145652
         30            Naftisol           65.841013
         35            Naftisol           69.176246
         45            Naftisol           73.867845

[1893 rows x 2 columns]

In [23]:
mean_df = stats_df.groupby(["Drug Regimen"]).agg(np.mean)
mean_df

Tumor Volume (mm3)
Drug Regimen                    
Capomulin              40.675741
Ceftamin               52.591172
Infubinol              52.884795
Ketapril               55.235638
Naftisol               54.331565
Placebo                54.033581
Propriva               52.322552
Ramicane               40.216745
Stelasyn               54.233149
Zoniferol              53.236507

In [24]:
median_df = stats_df.groupby(["Drug Regimen"]).agg(np.median)
median_df

Tumor Volume (mm3)
Drug Regimen                    
Capomulin              41.557809
Ceftamin               51.776157
Infubinol              51.820584
Ketapril               53.698743
Naftisol               52.509285
Placebo                52.288934
Propriva               50.854632
Ramicane               40.673236
Stelasyn               52.431737
Zoniferol              51.818479

In [27]:
variance_df = stats_df.groupby(["Drug Regimen"]).agg(np.var)
variance_df

Tumor Volume (mm3)
Drug Regimen                    
Capomulin              24.947764
Ceftamin               39.290177
Infubinol              43.128684
Ketapril               68.553577
Naftisol               66.173479
Placebo                61.168083
Propriva               42.351070
Ramicane               23.486704
Stelasyn               59.450562
Zoniferol              48.533355

In [29]:
std_df = stats_df.groupby(["Drug Regimen"]).agg(np.std)
std_df

Tumor Volume (mm3)
Drug Regimen                    
Capomulin               4.994774
Ceftamin                6.268188
Infubinol               6.567243
Ketapril                8.279709
Naftisol                8.134708
Placebo                 7.821003
Propriva                6.507770
Ramicane                4.846308
Stelasyn                7.710419
Zoniferol               6.966589

## Bar plots

This really means: generate a bar chart showing total number of mice per drug across all time points. You are simply doing a sum per drug, and you do not need to break it down by time point

In [13]:
# Generate a bar plot showing number of data points for each treatment regimen using pandas

In [14]:
# Generate a bar plot showing number of data points for each treatment regimen using pyplot

## Pie plots

In [15]:
# Generate a pie plot showing the distribution of female versus male mice using pandas

In [16]:
# Generate a pie plot showing the distribution of female versus male mice using pyplot

## Quartiles, outliers and boxplots

In [17]:
# Calculate the final tumor volume of each mouse across four of the most promising treatment regimens. Calculate the IQR and quantitatively determine if there are any potential outliers. 

In [18]:
# Generate a box plot of the final tumor volume of each mouse across four regimens of interest

## Line and scatter plots

Instead of picking any mouse, you will use mouse ID l509  for the line and scatter plots

In [19]:
# Generate a line plot of time point versus tumor volume for a mouse treated with Capomulin

In [20]:
# Generate a scatter plot of mouse weight versus average tumor volume for the Capomulin regimen

In [21]:
# Calculate the correlation coefficient and linear regression model for mouse weight and average tumor volume for the Capomulin regimen